src/buffer.cc

```diff
@@ -15,25 +15,19 @@
 #include "utils.hh"
 #include "window.hh"
 
+#include <cstring>
 #include <algorithm>
 
 namespace Kakoune
 {
 
-struct ParsedLines
+static BufferLines parse_lines(StringView data, ByteOrderMark* bom, EolFormat* eolformat)
```

Comment:
why change that signature ? That now requires to create those objects before.

Reply:
I need to know the BOM and EOL format every a reload is checked for, the only thing that's temporarily needed after this function returns is the `BufferLines` object.

Reply:
those are enums, they are extremely cheap to copy around, there is no need to pass them by pointer to the function, we can just copy the ones from ParsedLines. Moreover, they are stored in options, and I am not convinced they should be duplicated in the buffer.